src/Microsoft.AspNetCore.SystemWebAdapters/Microsoft.AspNetCore.SystemWebAdapters.csproj

```diff
@@ -23,27 +23,28 @@
   </ItemGroup>
 
   <ItemGroup>
-    <FrameworkFiles Include="**/*.Framework.cs" />
-    <StandardFiles Include="**/*.Standard.cs" />
+    <Compile Remove="Ref.Standard.cs" />
+    <Compile Remove="TypeForwards.Framework.cs" />
+  </ItemGroup>
+
+  <!-- Files in the project will be by default ASP.NET Core only. However, suffixes can change that:
+
+       - *.Framework.cs: These will only be in the .NET Framework build
+       - *.Shared.cs: These will be available in both
+  -->
+  <ItemGroup>
     <SharedFiles Include="**/*.Shared.cs" />
+    <FrameworkFiles Include="**/*.Framework.cs" />
   </ItemGroup>
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
     <Compile Remove="**/*" />
-    <Compile Include="@(StandardFiles)" />
-    <Compile Include="@(SharedFiles)" />
-
-    <!-- RemoteAppExtensions has a different namespace in net6.0 and netfx, so it can't be
-         included in .NET Standard builds. -->
-    <Compile Remove="Adapters/RemoteAppExtensions.Shared.cs" />
-    
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="6.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Options" Version="6.0.0" />
+    <Compile Include="Properties/PropertyInfo.cs" />
+    <Compile Include="Ref.Standard.cs" />
```

Comment:
Just a thought - since the source files for the netstandard target as completely disjoint with the source for other targets, should we just have a separate csproj for the netstandard build? I don't have a strong preference either way, but it seems worth considering since the inputs are so different now.

Reply:
How would that reconcile the type identities?

Reply:
We'd need to have the netstandard project/assembly have the same identity as the .NET 6 one. You'd just have one project that builds the netstandard version of the project and one that builds the net6/netfx ones. That does complicate the packaging story, though, so it's probably fine as it is.